### ISSUE TITLE
Sync One-way error during execution

### ISSUE THREAD

**Issue by Boia11** [abraunegg/onedrive#408]
### Bug Report Details ###
**Describe the bug**
```
Uploading file Rock64_Backup/Seagate-NC/nc-data/updater-xyzabcd/backups/nextcloud-15.0.1.2/apps/user_ldap/js/wizard/controller.js ...
Uploading 100% |oooooooooooooooooooooooooooooooooooooooo|   ETA   --:--:--:
 done.
Uploading file Rock64_Backup/Seagate-NC/nc-data/updater-xyzabcd/backups/nextcloud-15.0.1.2/apps/user_ldap/js/wizard/wizardDetectorFilterLogin.js ...
Giving up on sync after three attempts: Key not found: nextExpectedRanges
std.json.JSONException@std/json.d(494): Key not found: nextExpectedRanges
----------------
??:? object.Throwable.TraceInfo core.runtime.defaultTraceHandler(void*) [0x55831c6c1f]
??:? [0x55831be563]
??:? [0x55831b1687]
??:? [0x5583169787]
??:? [0x55831707db]
upload.d:114 [0x55830f5ddb]
upload.d:45 [0x55830f598b]
sync.d:1608 [0x558310358b]
sync.d:1384 [0x5583101a2f]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1375 [0x55831018d3]
sync.d:1056 [0x5583100ce7]
main.d:621 [0x5583114047]
main.d:483 _Dmain [0x5583111fcb]
```

**Application and Operating System Details:**
*   OS: `Linux Debian 4.4.132-1092-rockchip-ayufan-ge099fe1c6ac5 #1 SMP Thu Aug 9 19:41:01 UTC 2018 aarch64 GNU/Linux`
*   OneDrive Account Type: `OneDrive Business`
*   DMD or LDC compiler version: `ldc2-1.11.0-linux-aarch64`
*   Application configuration: 
```
onedrive version                    = v2.2.5-29-g460596d
Config path                         = /root/.config/onedrive
Config file found in config path    = true
Config option 'sync_dir'            = /root/OneDrive
Config option 'skip_file'           = ~*
Config option 'skip_symlinks'       = false
Config option 'monitor_interval'    = 3600
Config option 'min_notif_changes'   = 5
Config option 'log_dir'             = /var/log/onedrive/
Selective sync configured           = true
sync_list contents:
Rock64_Backup
```

*   Curl Version: 
```
curl 7.52.1 (aarch64-unknown-linux-gnu) libcurl/7.52.1 OpenSSL/1.0.2r zlib/1.2.8 libidn2/0.16 libpsl/0.17.0 (+libidn2/0.16) libssh2/1.7.0 nghttp2/1.18.1 librtmp/2.3
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL
```

**To Reproduce**
Steps to reproduce the behavior if not causing an application crash:
`onedrive --synchronize --upload-only --single-directory 'Rock64_Backup'`

**Comment by abraunegg** [comment]
@Boia11 
Unable to reproduce this using the latest 'master'

Can you:
* Rebuild with latest master
* Remove resume_upload file from `~/.config/onedrive/`
* Retry upload

Before the failed upload - was there any Internet connectivity issue?

**Comment by abraunegg** [comment]
@Boia11 
Can you also provide the **full** log for the entire sync process if this is still failing, not just the last few lines of the crash

**Comment by abraunegg** [comment]
@Boia11 
Whilst I cannot reproduce this, please can you rebuild your client with the following PR:
```
git clone https://github.com/abraunegg/onedrive.git
cd onedrive
git fetch origin pull/409/head:pr409
git checkout pr409
make
```
If you can test that for this issue that would be greatly appreciated

**Comment by abraunegg** [comment]
@Boia11 
Closing issue as code fix in v2.2.6 release
